src/Shared/LinkerFlags.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Internal
+{
+    internal static class LinkerFlags
+    {
+        /// <summary>
+        /// Flags for a member that is JSON (de)serialized.
+        /// </summary>
+        public const DynamicallyAccessedMemberTypes JsonSerialized = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.PublicProperties;
+
+        /// <summary>
+        /// Flags for a component
+        /// </summary>
+        public const DynamicallyAccessedMemberTypes Component = DynamicallyAccessedMemberTypes.All;
```

Comment:
This looks a bit drastic. Do you really need nested types or private fields?

Reply:
Blazor relies on being able to reflect on private properties on Components. @vitek-karas suggested using `All` as the most reliable way to preserve these.